### DISCUSSION TITLE
ApplicationInsights package

### DISCUSSION THREAD

**Discussion by fgheysels** [arcus-azure/arcus.observability#353] [General]
In `Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights`, we're making use of the `Microsoft.ApplicationInsights` package.
However, there also exist a `Microsoft.ApplicationInsights.AspnetCore` package.

It is not clear to me which one we should actually use ?

Any insights (pun intended) @stijnmoreels  @gverstraete

**Comment by stijnmoreels** [comment]
As we don't need HTTP functionality (middleware tracking, header extraction...) that is available in the `...AspNetCore` variant, and we want to be as general as possible here as it's also used in non-HTTP scenario's (like our Messaging and our Azure Functions templates) we should not be related to anything web or HTTP here. That would be for a more concrete library like Arcus web API and not an abstraction library as is observability.
Do you think we need this `...AspNetCore` variant, maybe?

**Reply by fgheysels** [reply]
No, was just wondering.
